jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -1417,8 +1417,19 @@ public void addListener(String className)
 
             ListenerHolder holder = getServletHandler().newListenerHolder(Source.JAVAX_API);
             holder.setListener(t);
-            getServletHandler().addListener(holder);
             addProgrammaticListener(t);
+            getServletHandler().addListener(holder);
+            if (_startListeners)
```

Comment:
A comment here would be helpful.... to say that this is here to eagerly start listeners added during the call to other listeners, so that they may process events (eg setAttributes) triggered in listeners themselves.